src/main/java/longah/TransactionList.java

```diff
@@ -0,0 +1,95 @@
+package longah;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a list of transactions.
+ */
+public class TransactionList {
+    private List<Transaction> transactions;
+
+    /**
+     * Constructs a new TransactionList instance.
+     */
+    public TransactionList() {
+        this.transactions = new ArrayList<>();
+    }
+
+    /**
+     * Adds a transaction to the list.
+     *
+     * @param transaction The transaction to add.
+     */
+    public void add(Transaction transaction) {
+        transactions.add(transaction);
+    }
+
+    /**
+     * Removes a transaction from the list by index.
+     *
+     * @param index The index of the transaction to remove.
+     */
+    public void remove(int index) {
+        if (index >= 0 && index < transactions.size()) {
+            transactions.remove(index);
+        } else {
+            System.out.println("Invalid index.");
+        }
+    }
+
+    /**
+     * Calculates the balances between members.
+     *
+     * @return A map containing the balances between members.
+     */
+    public Map<String, Double> calculateBalances() {
```

Comment:
Might be easier to imlpement this using `balance` attribute from `Member`, but I can work on this since I have a better idea of what I expect the algo might look like.